# Supermarket Transaction Analysis

-----

## 1. Exploring the data:

In [2]:
--Select top 10 rows from pos_operator_logs and pos_transactions
SELECT TOP 10 *
FROM pos_operator_logs

SELECT TOP 10 *
FROM pos_transactions;

(10 rows affected)

(10 rows affected)

Total execution time: 00:00:00.017

id Workstation_Group_ID Workstation_ID begin_date_time operator_id 
 1 8 19 2019-02-13 05:37:55.0000000 269 
 2 8 18 2019-02-13 05:37:55.0000000 268 
 3 8 17 2019-02-13 05:38:43.0000000 267 
 4 1 4 2019-02-13 07:01:26.0000000 332 
 5 1 7 2019-02-13 07:01:57.0000000 10 
 6 1 7 2019-02-13 07:02:16.0000000 10 
 7 1 4 2019-02-13 07:03:54.0000000 332 
 8 1 4 2019-02-13 07:05:52.0000000 332 
 9 1 7 2019-02-13 07:32:51.0000000 114 
 10 1 7 2019-02-13 07:35:12.0000000 114

id WorkstationGroupID begin_date_time end_date_time OperatorID basket_size t_cash t_card amount 
 1 1 2017-12-07 14:23:23.0000000 2017-12-07 14:24:36.0000000 101 23 1 0 112.71 
 2 1 2017-12-07 14:25:09.0000000 2017-12-07 14:27:00.0000000 101 29 1 0 54.76 
 3 1 2017-12-07 14:27:28.0000000 2017-12-07 14:27:48.0000000 101 3 1 0 14.77 
 4 1 2017-12-07 14:28:04.0000000 2017-12-07 14:28:29.0000000 101 12 1 0 37.88 
 5 1 2017-12-07 14:29:40.0000000 2017-12-07 14:30:32.0000000 101 7 1 0 115.34 
 6 1 2017-12-07 14:31:03.0000000 2017-12-07 14:32:01.0000000 101 12 0 1 232.89 
 7 1 2017-12-07 14:32:25.0000000 2017-12-07 14:36:03.0000000 101 36 1 0 183.15 
 8 1 2017-12-07 14:36:32.0000000 2017-12-07 14:37:15.0000000 101 10 1 0 39.68 
 9 1 2017-12-07 14:37:31.0000000 2017-12-07 14:38:07.0000000 101 15 1 0 44.53 
 10 1 2017-12-07 14:38:21.0000000 2017-12-07 14:39:12.0000000 101 15 0 1 42.73

|id|Workstation_Group_ID|Workstation_ID|begin_date_time|operator_id|
|---|---|---|---|---|
|1|8|19|2019-02-13 05:37:55.0000000|269|
|2|8|18|2019-02-13 05:37:55.0000000|268|
|3|8|17|2019-02-13 05:38:43.0000000|267|
|4|1|4|2019-02-13 07:01:26.0000000|332|
|5|1|7|2019-02-13 07:01:57.0000000|10|
|6|1|7|2019-02-13 07:02:16.0000000|10|
|7|1|4|2019-02-13 07:03:54.0000000|332|
|8|1|4|2019-02-13 07:05:52.0000000|332|
|9|1|7|2019-02-13 07:32:51.0000000|114|
|10|1|7|2019-02-13 07:35:12.0000000|114|

<br/>

|id|WorkstationGroupID|begin_date_time|end_date_time|OperatorID|basket_size|t_cash|t_card|amount|
|---|---|---|---|---|---|---|---|---|
|1|1|2017-12-07 14:23:23.0000000|2017-12-07 14:24:36.0000000|101|23|1|0|112.71|
|2|1|2017-12-07 14:25:09.0000000|2017-12-07 14:27:00.0000000|101|29|1|0|54.76|
|3|1|2017-12-07 14:27:28.0000000|2017-12-07 14:27:48.0000000|101|3|1|0|14.77|
|4|1|2017-12-07 14:28:04.0000000|2017-12-07 14:28:29.0000000|101|12|1|0|37.88|
|5|1|2017-12-07 14:29:40.0000000|2017-12-07 14:30:32.0000000|101|7|1|0|115.34|
|6|1|2017-12-07 14:31:03.0000000|2017-12-07 14:32:01.0000000|101|12|0|1|232.89|
|7|1|2017-12-07 14:32:25.0000000|2017-12-07 14:36:03.0000000|101|36|1|0|183.15|
|8|1|2017-12-07 14:36:32.0000000|2017-12-07 14:37:15.0000000|101|10|1|0|39.68|
|9|1|2017-12-07 14:37:31.0000000|2017-12-07 14:38:07.0000000|101|15|1|0|44.53|
|10|1|2017-12-07 14:38:21.0000000|2017-12-07 14:39:12.0000000|101|15|0|1|42.73|


In [3]:
SELECT 
MIN(begin_date_time) AS start_time, 
MAX(end_date_time) AS end_time
FROM pos_transactions;

(1 row affected)

Total execution time: 00:00:00.131

start_time end_time 
 2017-12-07 06:26:26.0000000 2019-04-10 22:08:55.0000000

|start_time|end_time|
|---|---|
|2017-12-07 06:26:26.0000000|2019-04-10 22:08:55.0000000|

- Data is stored from 2017-12-07 to 2019-04-10.

In [4]:
SELECT
YEAR(end_date_time) AS year_No,
ROUND(SUM(amount), 2) AS year_total_amount
FROM pos_transactions
GROUP BY YEAR(end_date_time)
ORDER BY year_No;

(2 rows affected)

Total execution time: 00:00:00.064

year_No year_total_amount 
 2017 5055633.62 
 2019 6802821.39

|year_No|year_total_amount|
|---|---|
|2017|5055633.62|
|2019|6802821.39|


- The supermarket operated in 2017 and 2019, closed in 2018.

- Total revenue for two years is about $11858454,&nbsp;in which revenue in 2019 is higher than 2017.

## 2. Transactions method:

In [7]:
WITH card_count AS (
SELECT COUNT(*) AS Card_transaction
FROM pos_transactions
WHERE t_card = 1)
,
cash_count AS (
SELECT COUNT(*) AS Cash_transaction
FROM pos_transactions
WHERE t_cash = 1)

SELECT *
FROM card_count
CROSS JOIN cash_count;

(1 row affected)

Total execution time: 00:00:00.050

Card_transaction Cash_transaction 
 78246 84487

|Card_transaction|Cash_transaction|
|---|---|
|78246|84487|

- Customers prefer to pay by cash for purchases rather than by card.

<br/>

+ Suggestion:

 --> Shop logistics should arrange more cash registers.

## 3. Amount spent per transaction using card versus cash:

In [8]:
SELECT
ROUND(AVG(CASE WHEN t_card = 1 AND t_cash = 0 THEN amount END),2) AS avg_card_trans,
ROUND(AVG(CASE WHEN t_cash = 1 AND t_card = 0 THEN amount END),2) AS avg_cash_trans
FROM pos_transactions;

Warning: Null value is eliminated by an aggregate or other SET operation.

(1 row affected)

Total execution time: 00:00:00.059

avg_card_trans avg_cash_trans 
 86.84 58.72

|avg_card_trans|avg_cash_trans|
|---|---|
|86.84|58.72|

- Customers paying by card spend more than customers paying with cash ($86.84 vs $58.72 in average).

<br/>

+ Suggestion:

--> Arranging security on card payment.

--> Marketing to encourage shoppers to use cash payment.

## 4. Exploring Sunday shopping:

In [9]:
SELECT 
CAST(end_date_time AS DATE) AS Sunday_date, 
[date_name], 
ROUND(SUM(amount), 2) AS total_amount
FROM (SELECT 
      end_date_time,
      DATENAME(dw,end_date_time) AS date_name, 
      amount
      FROM pos_transactions
      WHERE DATENAME(dw,end_date_time) = 'Sunday') a
GROUP BY CAST(end_date_time AS DATE), date_name;

(4 rows affected)

Total execution time: 00:00:00.076

Sunday_date date_name total_amount 
 2017-12-10 Sunday 276343.6000000002 
 2017-12-17 Sunday 344041.41000000015 
 2019-02-24 Sunday 143025.41000000012 
 2019-03-31 Sunday 169698.48000000019

|Sunday_date|date_name|total_amount|
|---|---|---|
|2017-12-10|Sunday|276343.6|
|2017-12-17|Sunday|344041.41|
|2019-02-24|Sunday|143025.41|
|2019-03-31|Sunday|169698.48|


- Supermarket closed in Sundays of 2018.

- Supermarket sales on the last Sundays of 2017 are 2 times higher than those on the first Sundays of 2019.

## 5. Daily trends analysis:

In [11]:
WITH cte_a AS (
SELECT
DATEPART(WEEK, CONVERT(DATE,end_date_time)) AS Week_No,
CONVERT(DATE, end_date_time) AS end_date,
DATENAME(DW, CONVERT(DATE, end_date_time)) AS day_name,
COUNT(id) AS total_transaction,
SUM(amount) AS total_amount,
AVG(amount) AS avg_amount
FROM pos_transactions
WHERE YEAR(end_date_time) >= 2019
GROUP BY CONVERT(DATE, end_date_time)
)

SELECT 
day_name, 
ROUND(MAX(total_amount),2) AS max_amount, 
MAX(total_transaction) AS max_transaction
FROM cte_a
GROUP BY day_name
ORDER BY max_amount DESC, max_transaction;

(7 rows affected)

Total execution time: 00:00:00.072

day_name max_amount max_transaction 
 Saturday 430575.4 4955 
 Friday 362678.52 4476 
 Thursday 342443.86 4565 
 Wednesday 239650.56 3506 
 Monday 233179.88 3615 
 Tuesday 224894.17 3397 
 Sunday 169698.48 2324

|day_name|max_amount|max_transaction|
|---|---|---|
|Saturday|430575.4|4955|
|Friday|362678.52|4476|
|Thursday|342443.86|4565|
|Wednesday|239650.56|3506|
|Monday|233179.88|3615|
|Tuesday|224894.17|3397|
|Sunday|169698.48|2324|


- Customers shop a lot on Thursdays, Fridays, and Saturdays.

- Saturdays have the highest amount of purchases compared to the rest of the week.

- Sundays have the least amount of shopping.

<br/>

+ Suggestion:

--> Creating more promotions on Sundays.

--> Arranging more personnel on weekends.

## 6. Number of employees analysis:

In [14]:
DROP VIEW labor;

GO

CREATE VIEW labor AS 
SELECT
CONVERT(DATE, begin_date_time) AS Shopping_date,
DATENAME(DW, CONVERT(DATE, begin_date_time)) AS Weekday_name,
COUNT(DISTINCT OperatorID) AS total_labor
FROm pos_transactions
GROUP BY CONVERT(DATE, begin_date_time);

GO

SELECT *
FROM (SELECT 
     Weekday_name, 
     total_labor, 
     RANK() OVER (PARTITION BY Weekday_name ORDER BY total_labor DESC) AS Ranking
     FROM labor
     WHERE Weekday_name IN ('Thursday', 'Friday', 'Saturday', 'Sunday')
) b
WHERE Ranking = 1

Commands completed successfully.

Commands completed successfully.

(4 rows affected)

Total execution time: 00:00:00.059

Weekday_name total_labor Ranking 
 Friday 27 1 
 Saturday 28 1 
 Sunday 20 1 
 Thursday 29 1

|Weekday_name|total_labor|Ranking|
|---|---|---|
|Friday|27|1|
|Saturday|28|1|
|Sunday|20|1|
|Thursday|29|1|


- Although Saturdays are the days with the highest shopping volume, the number of employees is lower than Thursdays and Fridays. While Sundays have less shopping, supermarket has a lot of staff.</span>

<br/>

+ Suggestion:

--> Arranging more staff on Saturdays, reducing the number of staff on Sundays.

# Conclusion:

The shop's logistics department should increase the number of cash registers available to accommodate more customers. It is important to ensure proper security measures are in place for card payments. The marketing team should focus on promoting the use of cash payments among shoppers. Additionally, creating attractive promotions specifically for Sundays can help attract more customers on that day. To handle the higher customer traffic during weekends, it is advisable to schedule additional personnel. Specifically, more staff should be assigned on Saturdays, while reducing the number of staff on Sundays.